### PR TITLE
Ensure job IDs are unique across workers

### LINKED ISSUE #1132: Ensure workers generate unique job IDs (MX)
In MX we did something like `jobId = jobId | (localGroupId << 32);` in `UniqueJobId` to ensure that workers that concurrently execute task tracker queries do so with unique job IDs.

### LINKED ISSUE #1147: Repartition Joins May Fail From Workers If JobId Sequences Overlap
In MX, when the jobids are the same of repartition joins on different workers, the task tracker may fail. It is a bit difficult to reproduce this error but repeated runs of [this gist setup](https://gist.github.com/aamederen/2e17bf34e1d37a5ca70169337af0e46d) helps producing it.

The cause seems like jobs and task that have the same jobid/taskid but belong to different queries collide with each other. A solution could be ensuring that jobid sequences are set to unique ranges on workers (as in BIGSERIAL columns of MX tables).

The error messages that arise include but not limiting to the following:
```
WARNING:  could not open file "base/pgsql_job_cache/job_10001/task_000030": No such file or directory
```

```
WARNING:  could not send remote query "SELECT task_tracker_task_status(10001, 33)"
DETAIL:  Client error: another command is already in progress
```

```
WARNING:  could not send remote query "SELECT task_tracker_assign_task  (10001, 2147483647, 'SELECT task_tracker_cleanup_job(10001)')"
DETAIL:  Client error: another command is already in progress
```

```
WARNING:  could not assign cleanup query for job 10001 to node "localhost:3701"
ERROR:  failed to execute job 10001
```

```
WARNING:  could not find the worker task
DETAIL:  Task jobId: 10000 and taskId: 30
```

### PR DESCRIPTION
**Pull request by marcocitus**
Fixes #1147 
Fixes #1132

### CODE REVIEW

File: src/backend/distributed/planner/multi_physical_planner.c [hunk 1]
```diff
@@ -1723,8 +1725,15 @@ UniqueJobId(void)
 	jobIdDatum = DirectFunctionCall1(nextval_oid, sequenceIdDatum);
 	jobId = DatumGetInt64(jobIdDatum);
 
+	/*
+	 * Add the local group id information to the jobId to
+	 * prevent concurrent jobs on different groups to conflict.
+	 */
+	localizedJobId = jobId | (localGroupId << 32);
```
Comment (anarazel) [inline src/backend/distributed/planner/multi_physical_planner.c]: Not that it's that likely - but it's conceivable to extend 32bit in the jobid sequence.  Perhaps we should cast jobid to uint32 and simply % PG_UINT32_MAX jobid beforehand? It's really unlikely to have jobs from before / after 4 billion.    A test with such big ids won't hurt either...

**Comment by anarazel** [comment]
(not assigned reviewer, but besides that wraparound issue this looks good to me)

File: src/backend/distributed/planner/multi_physical_planner.c (hunk 1 above)
Comment (marcocitus) [inline src/backend/distributed/planner/multi_physical_planner.c]: The job ID sequence wraps around at 2^32.

Comment (marcocitus) [inline src/backend/distributed/planner/multi_physical_planner.c]: Oh. 2^32 is the maximum value, so it's actually off by one. Anyway, I'm inclined to just fix that by moving away from using a sequence.

**Pull request merged**